core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -304,6 +316,45 @@ class WorkspaceService(protected val userInfo: UserInfo,
     })
   }
 
+  private def getAzureCloudContextFromWorkspaceManager(workspaceContext: Workspace, parentSpan: Span = null) = {
+    workspaceContext.workspaceType match {
+      case WorkspaceType.McWorkspace => {
+        val span = startSpanWithParent("getWorkspaceFromWorkspaceManager", parentSpan)
+
+        try {
+          val wsmInfo = workspaceManagerDAO.getWorkspace(workspaceContext.workspaceIdAsUUID, userInfo.accessToken)
+
+          Option(wsmInfo.getAzureContext) match {
+            case Some(azureContext) => Some(WorkspaceAzureCloudContext(
+              azureContext.getTenantId,
+              azureContext.getSubscriptionId,
+              azureContext.getResourceGroupId)
+            )
+            case None => {
+                throw new RawlsExceptionWithErrorReport(errorReport = ErrorReport(
+                  StatusCodes.NotImplemented, s"Non-azure MC workspaces not supported [id=${workspaceContext.workspaceId}]"
+              ))
+            }
+          }
+        } catch {
+          case e: ApiException =>
+            logger.warn(s"Error retrieving MC workspace from workspace manager [id=${workspaceContext.workspaceId}, code=${e.getCode}]")
+
+            if (e.getCode == StatusCodes.NotFound.intValue) {
+              span.setStatus(Status.NOT_FOUND)
+              throw new RawlsExceptionWithErrorReport(errorReport = ErrorReport(StatusCodes.NotFound, e))
+            } else {
+              span.setStatus(Status.INTERNAL)
```

Comment:
Why do you want to convert all other codes to INTERNAL? Is there no way to preserve the original status code?

Reply:
No, these are opencensus specific [codes](https://opencensus.io/tracing/span/status/) that don't directly map to HTTP status codes.

Reply:
Sorry, my comment was more about the next line (348), where we lose the original status code in the `RawlsExceptionWithErrorReport`.

Reply:
Ah, good call, will fix.